sequence_processing_pipeline/NuQCJob.py

```diff
@@ -167,33 +171,73 @@ def run(self, callback=None):
+            completed_files = list(glob.glob(pattern))
 
-            if not self._get_failed_indexes(project_name, job_id):
-                raise PipelineError("QCJob did not complete successfully.")
-
-            # TODO: IMPLEMNENT A NEW FILTER FOR FILTERED FASTQ.GZ FILES THAT
-            #  ARE BELOW THE MINIMUM FILE SIZE THRESHOLD INTO A NEW FOLDER
-            #  NAMED 'ZERO-LENGTH-FILES'.
-
-            # determine where the filtered fastq files can be found and move
-            # the 'zero-length' files to another directory.
             if needs_human_filtering is True:
                 filtered_directory = join(source_dir, 'filtered_sequences')
             else:
                 filtered_directory = join(source_dir, 'trimmed_sequences')
 
-            if not exists(filtered_directory):
-                raise PipelineError(f"{filtered_directory} does not exist")
+            # create the properly named directory to move files to in
+            # in order to preserve legacy behavior.
+            makedirs(filtered_directory, exist_ok=True)
+
+            # get the list of sample-names in this project.
+            samples_in_project = [x[0] for x in self.sample_ids
+                                  if x[1] == project_name]
+
+            files_to_move = []
+            regex = re.compile(r'^(.*)_S\d_L\d\d\d_R\d+_\d\d\d\.fastq\.gz$')
+            for fp in completed_files:
+                file_name = basename(fp)
+                substr = regex.search(file_name)
+                if substr is None:
+                    raise ValueError(f"{file_name} does not follow naming "
+                                     " pattern.")
+                else:
+                    # check if found substring is a member of this
+                    # project. Note sample-name != sample-id
+                    if substr[1] in samples_in_project:
+                        files_to_move.append(fp)
+
+            for fp in files_to_move:
+                move(fp, filtered_directory)
 
+            # once fastq.gz files have been moved into the right project,
+            # we now need to consider the html and json fastp_reports
+            # files.
+            old_html_path = join(self.output_path, 'fastp_reports_dir', 'html')
+            old_json_path = join(self.output_path, 'fastp_reports_dir', 'json')
+
+            new_html_path = join(source_dir, 'fastp_reports_dir', 'html')
+            new_json_path = join(source_dir, 'fastp_reports_dir', 'json')
+
+            makedirs(new_html_path, exist_ok=True)
+            makedirs(new_json_path, exist_ok=True)
+
+            pattern = f"{old_html_path}/*.html"
+            completed_htmls = list(glob.glob(pattern))
+
+            for fp in completed_htmls:
+                move(fp, new_html_path)
+
+            pattern = f"{old_json_path}/*.json"
+            completed_jsons = list(glob.glob(pattern))
+
+            for fp in completed_jsons:
+                move(fp, new_json_path)
+
```

Comment:
Why move each of the files vs. moving the folders?

Reply:
Good catch, thanks! The new QC algorithm processes files across all projects, but we expect the individual fastq.gz and the html/json reports to be organized by project. Hence there's a framework to assess and move each individual file. Unfortunately the code to move only the files associated with a given project wasn't added. I've added it now and am running a short test.